docs/api-docs/storefront/graphql/graphql-products.md

```diff
@@ -0,0 +1,875 @@
+            url1280wide: url(width: 1280)
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+<!--
+type: tab
+title: Response
+-->
+
+```json title="Example response: Get product images at different resolutions" lineNumbers
+{
+  "data": {
+    "site": {
+      "product": {
+        "images": {
+          "edges": [
+            {
+              "node": {
+                "url320wide": "https://cdn11.bigcommerce.com/s-o8ertuc7vw/images/stencil/320w/products/113/379/IMAG0729__87810.1662649673.jpg",
+                "url640wide": "https://cdn11.bigcommerce.com/s-o8ertuc7vw/images/stencil/640w/products/113/379/IMAG0729__87810.1662649673.jpg",
+                "url960wide": "https://cdn11.bigcommerce.com/s-o8ertuc7vw/images/stencil/960w/products/113/379/IMAG0729__87810.1662649673.jpg",
+                "url1280wide": "https://cdn11.bigcommerce.com/s-o8ertuc7vw/images/stencil/1280w/products/113/379/IMAG0729__87810.1662649673.jpg"
+              }
+            }
+          ]
+        }
+      }
+    }
+  }
+}
+```
+
+<!-- type: tab-end -->
+
+## Get product metafields
+
+You can query the metafields for products. Only metafields that have storefront permissions are returned (i.e. permissions must be set to `write_and_sf_access` or `read_and_sf_access`).
+
+The following query retrieves product metafields for the specified product. The query retrieves only product metafields. See [Get variant metafields](/api-docs/storefront/graphql/variants#get-variant-metafields) to retrieve variant metafields. 
+
+<!--
+type: tab
+title: Query
+-->
+
+```graphql title="Example query: Get product metafields for a product" lineNumbers
+query {
+  site {
+    product (entityId: 113) {
+      metafields (namespace: "Warehouse Locations") {
+        edges {
+          node {
+            id
+            entityId
+            key
+            value
+          }
+        }
+      } 
+    }
+  }
+}
+```
+
+<!--
+type: tab
+title: Response
+-->
+
+```json title="Example response: Get product metafields for a product" lineNumbers
+{
+  "data": {
+    "site": {
+      "product": {
+        "metafields": {
+          "edges": [
+            {
+              "node": {
+                "id": "TWV0YWZpZWxkczoxNQ==",
+                "entityId": 15,
+                "key": "Location",
+                "value": "4HG"
+              }
+            },
+            {
+              "node": {
+                "id": "TWV0YWZpZWxkczoxNg==",
+                "entityId": 16,
+                "key": "Site",
+                "value": "2SL"
+              }
+            }
+          ]
+        }
+      }
+    }
+  }
+}
+```
+<!-- type: tab-end -->
+
+## Get product custom fields
+
+You can query the custom fields for products. The following example retrieves custom fields for the specified product:
+
+<!--
+type: tab
+title: Query
+-->
+
+```graphql title="Example query: Get custom fields for a product" lineNumbers
+query {
+  site {
+    product (entityId: 113) {
+      customFields {
```

Comment:
Should we maybe use pagination params explicitly `customFields(first: 10)` to educate our merchants and set correct expectations? (the same for all other list types with pagination).

It might also be a good idea to mention that if `first` is not specified it will return only 10 items by default

Reply:
@bc-dima-pasieka That sounds like a good idea. I just added the `first` pagination field for applicable queries. 

Quick question, is 10 the default for all queries that can be paginated? If so, I'll add a note to the [pagination section](https://developer.bigcommerce.com/api-docs/storefront/graphql/graphql-storefront-api-overview#pagination) of the GraphQL Storefront API overview page

Reply:
Will update the pagination section to say 10 is the default for all queries (screenshot attached)

![default is 10](https://github.com/bigcommerce/dev-docs/assets/96258747/9db721e1-dbd4-44db-bb02-c332fa72188c)